Digipost/APIClient+URLSessionTask.swift

```diff
@@ -109,17 +114,24 @@ extension APIClient {
             }, completionHandler: { (response, fileURL, error) -> Void in
                 if let actualError = error {
                     if (error.code != NSURLErrorCancelled) {
-                        failure(error: APIError(error: error))
+                        if self.isUnauthorized(response as? NSHTTPURLResponse) {
+                            OAuthToken.removeAcessTokenForOAuthTokenWithScope(kOauth2ScopeFull)
+                            self.validateFullScope {
+                                failure(error: APIError(domain: Constants.Error.apiClientErrorDomain, code: Constants.Error.Code.UnknownError.rawValue, userInfo: nil))
+                            }
+                        } else {
+                            failure(error: APIError(error: error))
+                        }
                     }
                 } else if let actualFileUrl = fileURL {
-                        success(url:actualFileUrl)
+                    success(url:actualFileUrl)
                 }
                 // we get here if the request was canceled, should do nothing.
         })
         return task
     }
 
-    func isUnauthorized(urlResponse: NSHTTPURLResponse?) -> Bool {
+    private func isUnauthorized(urlResponse: NSHTTPURLResponse?) -> Bool {
         if let actualResponse = urlResponse as NSHTTPURLResponse! {
             if (actualResponse.statusCode == 403 || actualResponse.statusCode == 401) {
```

Comment:
I think we should separate between 403 (Forbidden) and 401 (Unauthorized). 401 states that the user is not authorized (or rather: authenticated), 403 status that the user is authenticated, but not allowed to access the given resource. In this pull requests we want to be looking for 401.